src/OpenTelemetry.AutoInstrumentation.Native/cor_profiler.cpp

```diff
@@ -2494,14 +2494,47 @@ HRESULT CorProfiler::CallTarget_RewriterCallback(RejitHandlerModule* moduleHandl
             &wrapper_type_def);
         if (FAILED(hr) || wrapper_type_def == mdTypeDefNil)
         {
-            Logger::Warn("*** CallTarget_RewriterCallback() Failed for: ", caller->type.name, ".", caller->name,
+            Logger::Error("*** CallTarget_RewriterCallback() Failed for: ", caller->type.name, ".", caller->name,
```

Comment:
Should it not be a warning? It is an oddity, but the application and instrumentation would continue to work without any bad side-effects? Should the user react when he notices such an error?

Maybe we should even "accept" such behavior so that we could use e.g. `#if NETFRAMEWORK` on a whole instrumentation class? Then we could even change this level to e.g. debug.

Leaving it for a discussion + potentially separate PR.